src/input/input_python.rs

```diff
@@ -102,14 +97,6 @@ impl<'a> Input<'a> for PyAny {
         Some(self.getattr(name))
     }
 
-    fn input_is_instance(&self, class: &PyAny, _json_mask: u8) -> PyResult<bool> {
-        // See PyO3/pyo3#2694 - we can't use `is_instance` here since it requires PyType,
-        // and some check objects are not types, this logic is lifted from `is_instance` in PyO3
-        let result = unsafe { ffi::PyObject_IsInstance(self.as_ptr(), class.as_ptr()) };
-        py_error_on_minusone(self.py(), result)?;
-        Ok(result == 1)
-    }
```

Comment:
The comment doesn't seem to be true, I replaced this with calling the method on `&PyAny` and it works fine, no tests fail here or in Pydantic. I did notice some alleged benchmark regressions for code that is not touched by any other changes here as far as I can tell, so could this be a performance thing? If so I can re-add it (although personally I would happily give up what seems like a very small performance difference to not use unsafe).

Reply:
Ok the issue was simpler and had nothing to do with unsafe: the `JsonInput` version of this short-circuited, but by removing this I was causing the conversion from json -> python object -> check the type. I now added back the short circuit but via `matches!(extra.mode, InputType::Python)` which is not only more explicit but I suspect might be faster.